frontend/src/app/main/ui/ds/utilities/date.cljs

```diff
@@ -0,0 +1,41 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.main.ui.ds.utilities.date
+  (:require-macros
+   [app.common.data.macros :as dm]
+   [app.main.style :as stl])
+  (:require
+   [app.common.data :as d]
+   [app.main.ui.ds.foundations.typography :as t]
+   [app.main.ui.ds.foundations.typography.text :refer [text*]]
+   [app.util.time :as dt]
+   [rumext.v2 :as mf]))
+
+(defn valid-date?
+  [date]
+  (or (dt/datetime? date) (number? date)))
+
+(def ^:private schema:date
+  [:map
+   [:class {:optional true} :string]
+   [:as {:optional true} :string]
+   [:date [:fn valid-date?]]
+   [:selected {:optional true} :boolean]])
+
+(mf/defc date*
+  {::mf/props :obj
+   ::mf/schema schema:date}
+  [{:keys [class as date selected] :rest props}]
+  (let [as (or as "span")
+        class (d/append-class class (stl/css-case :date true :is-selected selected))
+        date (cond-> date (not (dt/datetime? date)) dt/datetime)]
+    [:> text* {:as as :typography t/body-medium :class class}
+     (dm/str
+      (dt/format date :date-full)
+      " . "
+      (dt/format date :time-24-simple)
+      "h")]))
```

Comment:
There's a proper HTML element for this: [`<time>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time) (and we should also set its `datetime` attribute)